Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -442,12 +442,17 @@ - (void)setValuesForKeysWithDictionary:(NSDictionary*)keyedValues {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSDictionary*)dictionaryWithValuesForKeys:(NSArray*)keys {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSDictionary* results = [NSMutableDictionary dictionary];
+
+    for (NSString* key in keys) {
+        id retValue = [self valueForKey:key];
+        [results setValue:((retValue != nil) ? retValue : [NSNull null]) forKey:key];
+    }
+
+    return results;
```

Comment:
If you pass `nil` for keys, does the reference platform return an empty dictionary?

Should the resulting dictionary be immutable (`[[results copy] autorelease]`)